File: dart/lib/bebop_handlers/session.dart
import 'package:lean_dart_client/classes/game_state.dart';

import 'bebop_handler.dart';
import 'package:lean_dart_client/schemas.g.dart';

class SessionMsgHandler extends BebopHandler {
  @override
  void handleMessage(sender, message) {
    message = SessionMsg.decode(message);
    GameState.instance.session = message;
    print(message.clientId);
    print(message.sessionId);
  }
}

File: typescript/src/bebop_handlers/session.ts
import GameState from '../classes/gameState';
import { SessionMsg } from '../models/pactheman.models';
import BebopHandler from './bebopHandler';

export default class SessionMsgHandler implements BebopHandler {
    handleMessage(sender: any, message: any): void {
        message = SessionMsg.decode(message);
        GameState.Instance.session = message;
        console.log(message.clientId);
        console.log(message.sessionId);
    }
}